### DISCUSSION TITLE
Adding specific columns for my inventory

### DISCUSSION THREAD

**Discussion by mudassar-mohammed** [inventree/InvenTree#6753] [Q&A] [answered]
Hi All,
I need some help because i don't usually work with code and a bit difficult for me to understand how things work and how to setup API's or plugins.

I have my own devices and i want to use inventree to keep stock of them and track them overall. My data to keep track of is not generic as the part columns in inventree currently and I want it to display some information such as firmware version, MAC ID etc in their own columns. Is it possible to create new columns that are displayed in the part list instead of the generic ones (revision, units)

**Comment by matmair** [comment]
This is not a built-in use case so you would need to adapt both backend and frontend code to make that happen. Have you looked into Snipe-It? They offer customisable columns.

**Reply by mudassar-mohammed** [reply]
I did take a look at Snipe-It. Its more on the lines of asset management whereas Im looking to use inventree to manage the inventory at different locations, sales of the devices to customers, BOM management, location tracking.

I just need to know if the backend and frontend can be optimised to include some custom columns that I would require. I am using inventree on my local server right now but I plan to send this to my software team who will then start a server so multiple users from my company can start using inventree. They will also look into the API's and if we already have something specific we can use to create custom columns, I can point them to it

**Comment by wolflu05** [comment, marked as answer]
That sounds to me like the part parameter table+templates feature. Have you had a look at https://docs.inventree.org/en/stable/part/parameter/ ?

**Reply by mudassar-mohammed** [reply]
Got it. Thank you!

I can use this to add specific information.
